quinn-proto/src/tests.rs

```diff
@@ -208,7 +206,10 @@ impl Pair {
 
     fn connect(&mut self) -> (ConnectionHandle, ConnectionHandle) {
         info!(self.log, "connecting");
-        let client_conn = self.client.connect(self.server.addr, "localhost").unwrap();
+        let client_conn = self
```

Comment:
It still seems to me the repetition that you introduce here points to a worse API. Do you agree that it would be more common for applications using Quinn to have a single `ClientConfig` that can be used across connections versus specific per-connection `ClientConfig`s?

An alternative here would be to have another `connect()` method on `Endpoint` that allows passing in a `ClientConfig` explicitly. I feel that is better because it provides a streamlined, simple API for common use cases and a more complex API for those who need more power/expressivity.

Reply:
I totally agree that this is a niche case, and an API that sets a default config seems reasonable. That would belong in the high-level API crate, though, not -proto, right?

Reply:
Latest revision includes this in the high-level API.